src/main/java/com/box/sdk/BoxAPIResponseException.java

```diff
@@ -40,16 +40,12 @@ public BoxAPIResponseException(String message, BoxAPIResponse responseObj) {
         try {
             responseJSON = JsonObject.readFrom(responseObj.bodyToString());
 
-            if (responseObj.bodyToString() != null && responseJSON.get("request_id") != null
-                    || this.getHeaders().containsKey("BOX-REQUEST-ID")) {
-
-                if (responseJSON.get("request_id") != null) {
-                    requestId += responseJSON.get("request_id").asString();
-                }
+            if (responseObj.bodyToString() != null && responseJSON.get("request_id") != null) {
```

Comment:
What happens if `responseObj.bodyToString()` is null?  Wouldn't the statement above have already thrown in that case?  It might be helpful to have a test case with an empty body and the `BOX-REQUEST-ID` header to exercise that code path

Reply:
What is the expected behavior with empty body? The statement above does throw and we exit into the catch where we present the user with the message

`The API returned an error code <ERROR CODE>`

Wondering if that is the message we expect when there is an overall error reading the error response body or should we append the request id from the header into that error message as well? So something like this:

`The API returned an error code <ERROR CODE> | <REQUEST ID>`

Thank you @mattwiller !